pkgs/servers/web-apps/snipe-it/default.nix

```diff
@@ -13,6 +13,7 @@ let
       ln -s ${dataDir}/public/uploads $out/public/uploads
       ln -s ${dataDir}/bootstrap/cache $out/bootstrap/cache
       chmod +x $out/artisan
+      substituteInPlace config/database.php --replace "/usr/local/bin" "${mariadb}/bin"
```

Comment:
This should be done in postPatch

Reply:
> This should be done in postPatch.

Doesn't work as $NIX_BUILD_TOP is empty during that phase:

```nix
    postPatch = ''
      echo 'ls -la $NIX_BUILD_TOP'
      ls -la $NIX_BUILD_TOP
      substituteInPlace config/database.php --replace "env('DB_DUMP_PATH', '/usr/local/bin')" "env('DB_DUMP_PATH', '${mariadb}/bin')"
    '';
```

building:
```
$ nix --log-format bar-with-logs build -f . snipe-it
snipe-it> unpacking sources
snipe-it> patching sources
snipe-it> ls -la $NIX_BUILD_TOP
snipe-it> total 48
snipe-it> drwx------ 1 nixbld nixbld    16 Aug  9 14:43 .
snipe-it> drwxr-x--- 1 nobody nixbld    54 Aug  9 14:43 ..
snipe-it> -rw-r--r-- 1 nixbld nixbld 48838 Aug  9 14:43 env-vars
snipe-it> substitute(): ERROR: file 'config/database.php' does not exist
error: builder for '/nix/store/8m7w0b82d3z9scfc18131nz54wsqp479-snipe-it.drv' failed with exit code 1;
       last 8 log lines:
       > unpacking sources
       > patching sources
       > ls -la $NIX_BUILD_TOP
       > total 48
       > drwx------ 1 nixbld nixbld    16 Aug  9 14:43 .
       > drwxr-x--- 1 nobody nixbld    54 Aug  9 14:43 ..
       > -rw-r--r-- 1 nixbld nixbld 48838 Aug  9 14:43 env-vars
       > substitute(): ERROR: file 'config/database.php' does not exist
       For full logs, run 'nix log /nix/store/8m7w0b82d3z9scfc18131nz54wsqp479-snipe-it.drv'.
```